### PR TITLE
refactor(lsp): changes for lsp_types 0.97.0

### PR DESCRIPTION
**Pull request by nayeemrmn**
~~Blocked~~ by https://github.com/ebkalderon/tower-lsp/pull/401. But currently this PR patches that revision in `Cargo.toml` so this could be landed.

Unblocks jupyter cell continuity: https://github.com/denoland/vscode_deno/issues/932.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -43,6 +43,11 @@ edition = "2021"
 license = "MIT"
 repository = "https://github.com/denoland/deno"
 
+# Until https://github.com/ebkalderon/tower-lsp/pull/401 gets landed, we're
+# patching it in.
+[patch.crates-io]
+tower-lsp = { git = "https://github.com/denoland/tower-lsp.git", rev = "910e0f8" }
```
Comment (bartlomieju) [inline Cargo.toml:49]: Could you move it to the very bottom? When we're trying `deno_core` locally `[patch]` is usually placed at the bottom

File: cli/lsp/capabilities.rs [hunk 2]
```diff
@@ -154,5 +154,8 @@ pub fn server_capabilities(
     // TODO(nayeemrmn): Support pull-based diagnostics.
     diagnostic_provider: None,
     inline_value_provider: None,
+    inline_completion_provider: None,
+    // TODO(nayeemrmn); Provide!
```
Comment (bartlomieju) [inline cli/lsp/capabilities.rs:158]: I assume this is gonna happen in the follow up PR?

**Review by bartlomieju — COMMENTED** [review]
Looks reasonable, I'll wait for David's approval

File: cli/lsp/capabilities.rs (hunk 2 above)
Comment (nayeemrmn) [reply cli/lsp/capabilities.rs]: Yeah, though I hadn't decided that when writing that. Removed the comment.

File: Cargo.toml [hunk 3]
```diff
@@ -394,3 +394,8 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.zstd-sys]
 opt-level = 3
+
+# Until https://github.com/ebkalderon/tower-lsp/pull/401 gets landed, we're
+# patching it in.
+[patch.crates-io]
+tower-lsp = { git = "https://github.com/denoland/tower-lsp.git", rev = "910e0f8" }
```
Comment (dsherret) [inline Cargo.toml:401]: We can't publish to crates.io with this. Are we just not going to publish with this PR for now?

**Review by dsherret — APPROVED** [review]
LGTM, but we can't publish to crates.io after landing this pr (which may or may not be a problem)

File: Cargo.toml (hunk 3 above)
Comment (bartlomieju) [reply Cargo.toml:401]: Why can't we publish to crates.io with this one?

Comment (dsherret) [reply Cargo.toml:401]: IIRC it will error because all the dependencies aren't on crates.io (and I would really hope it has this behaviour)

Comment (bartlomieju) [reply Cargo.toml:401]: Got it, then we should publish our fork to crates.io

Comment (bartlomieju) [reply Cargo.toml:401]: Published as https://crates.io/crates/deno_tower_lsp

**Pull request merged**